hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/ActionStack.java

```diff
@@ -192,12 +191,13 @@ private ContractAction finalFormOf(@NonNull final ContractAction action, @NonNul
                 } else {
                     builder.output(tuweniToPbjBytes(frame.getOutputData()));
                     if (action.targetedAddress() != null) {
-                        final var lazyCreatedAddress = pbjToBesuAddress(action.targetedAddressOrThrow());
-                        try {
-                            builder.recipientAccount(accountIdWith(hederaIdNumberIn(frame, lazyCreatedAddress)));
-                        } catch (final IllegalArgumentException | NullPointerException | ArithmeticException e) {
-                            // handle non-existing to/receiver address
-                            builder.recipientAccount((AccountID) null);
+                        final var maybeCreatedAddress = pbjToBesuAddress(action.targetedAddressOrThrow());
+                        final var maybeCreatedAccount = proxyUpdaterFor(frame).getHederaAccount(maybeCreatedAddress);
+                        // Fill in the account of id of a successful lazy creation; but just leave
+                        // the targeted address in case of a failed lazy-creation or a call to a
+                        // non-existent address
```

Comment:
How does this fix
> A previewnet transaction [here](https://hashscan.io/previewnet/transaction/1712267533.938791678) generated a CONTRACT_ACTION sidecar whose recipient was a recipient_account with an out-of-range accountNum.

Or is the out-of-range accountNum referred to the `((AccountID)null)` that was in the previous code?

Reply:
The previous code would always set a long-zero address as the `recipient_account`, **even if** that long-zero address didn't exist.

So e.g. `0xE8D4A50FFF` would become `receipient_account { account_num: 999999999999 }`.